src/Features/CSharp/Portable/Formatting/CSharpNewDocumentFormattingService.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.Formatting
+{
+    [ExportLanguageService(typeof(INewDocumentFormattingService), LanguageNames.CSharp)]
+    [Shared]
+    internal class CSharpNewDocumentFormattingService : AbstractNewDocumentFormattingService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpNewDocumentFormattingService([ImportMany] IEnumerable<Lazy<INewDocumentFormattingProvider, LanguageMetadata>> providers)
+            : base(providers)
+        {
+        }
+
+        protected override string Language => LanguageNames.CSharp;
```

Comment:
interesting.  do you need this given that the service was already exported for C# anyways?

Reply:
also, dumb question, but does your system make sure that if it *can't* get an INewDocumentFormattingService that that's ok?  we need to make sure we don't break F#/TS

Reply:
> do you need this

This is needed to filter the providers by metadata in the base class. If they were filtered at this level, it would have to be in the `base(..)` constructor call, which would defeat the purpose of the lazy (I think?)